src/state/state.ts

```diff
@@ -0,0 +1,15 @@
+import { cloneDeep } from 'lodash';
+import { Config } from '../config/schema';
+
+type State = {
+  config: Config;
+  dataFetcherInterval?: NodeJS.Timeout;
+};
+
+let state: State | undefined;
+
+export const getState = (): State => cloneDeep(state)!;
```

Comment:
Yeah, at the start of the app the state is undefined. It could I guess also be initialised with an empty state but that's a lot more verbose than an `!` 🤷  or the state could be initialised as an object and then given as an argument to consuming functions.

Reply:
I'd avoid clone deep and just do shallow state updates.

> Yeah, at the start of the app the state is undefined.

What I usually do is throw an error in the getter if the value is not defined. The expectation is that the state is initalized at the start of the app, but potentially needs some values (e.g. config).